pymatgen/electronic_structure/core.py

```diff
@@ -27,7 +27,15 @@ def __float__(self) -> float:
         return float(self.value)
 
     def __str__(self) -> str:
-        return str(self.value)
+        return str(self.name)
+
+    @classmethod
+    def from_str(cls, val: str) -> Spin:
+        """Returns Spin from dict representation."""
+        value = {"spin.up": 1, "up": 1, "spin.down": -1, "down": -1}.get(val.lower(), val)
+        if value not in (1, -1, "1", "-1"):
+            raise ValueError(f"Invalid Spin {value=}")
```

Comment:
Can we change this to `__init__` and something like below:

```py
    def __init__(self, val) -> Spin:
         """Returns Spin from possible values."""
         value = {"spin.up": 1, "up": 1, "spin.down": -1, "down": -1}
         if val in {-1, 1}:
                  raise DeprecationWarning('Int initializing is deprecated, please use "up"/"down" instead')
                  return Spin(int(value))
         if val in value:
                  return Spin(int(value[val]))

         raise ValueError(f"Invalid Spin {value=}")

```

Reply:
This might prevent it from becoming breaking change. Thus people you have `Spin(str(1))` in their codes(workflows/scripts) will correctly initialize the  `Spin` and also leave a warning for the future.